### PR TITLE
perf(core): persist workspaces in parallel

### PR DESCRIPTION
**Pull request by merceyz**
**What's the problem this PR addresses?**

Yarn persists workspaces one at a time which with 1000s of workspaces noticeably slows down an install.

**How did you fix it?**

Persist the workspaces in parallel.

**Results**

Tested on the [DefinitelyTyped repo](https://github.com/jakebailey/DefinitelyTyped/tree/b09f694f323b0c5aa80c97401815f7741903be77) with its 8629 workspaces (matching only `types/*`):

```console
$ YARN_IGNORE_PATH=1 hyperfine -w 1 "node ./after-plimit.cjs --mode skip-build" "node ./after.cjs --mode skip-build" "node ./before.cjs --mode skip-build"
Benchmark 1: node ./after-plimit.cjs --mode skip-build
  Time (mean ± σ):     16.106 s ±  0.605 s    [User: 13.252 s, System: 9.803 s]
  Range (min … max):   15.090 s … 17.147 s    10 runs

Benchmark 2: node ./after.cjs --mode skip-build
  Time (mean ± σ):     16.644 s ±  0.635 s    [User: 13.874 s, System: 10.015 s]
  Range (min … max):   15.932 s … 18.146 s    10 runs

Benchmark 3: node ./before.cjs --mode skip-build
  Time (mean ± σ):     17.390 s ±  0.641 s    [User: 13.684 s, System: 10.219 s]
  Range (min … max):   16.582 s … 19.000 s    10 runs

Summary
  'node ./after-plimit.cjs --mode skip-build' ran
    1.03 ± 0.06 times faster than 'node ./after.cjs --mode skip-build'
    1.08 ± 0.06 times faster than 'node ./before.cjs --mode skip-build'
```

**Checklist**
- [x] I have read the [Contributing Guide](https://yarnpkg.com/advanced/contributing).
- [x] I have set the packages that need to be released for my changes to be effective.
- [x] I will check that all automated PR checks pass before the PR gets reviewed.

### CODE REVIEW

File: packages/yarnpkg-core/sources/Project.ts [hunk 1]
```diff
@@ -1880,11 +1880,11 @@ export class Project {
...
-    await this.persistLockfile();
-
-    for (const workspace of this.workspacesByCwd.values()) {
-      await workspace.persistManifest();
-    }
+    const limit = pLimit(4);
+    await Promise.all([
+      this.persistLockfile(),
+      Promise.all(this.workspaces.map(workspace => limit(() => workspace.persistManifest()))),
+    ]);
```
Comment (arcanis) [inline packages/yarnpkg-core/sources/Project.ts:1887]: Perhaps a spread to make it more readable?
```suggestion
    await Promise.all([
      this.persistLockfile(),
      ...this.workspaces.map(workspace => {
        return limit(() => workspace.persistManifest());
      }),
    ]);
```

Comment (merceyz) [reply packages/yarnpkg-core/sources/Project.ts:1887]: I wanted to avoid having to iterate the array when `Promise.all` could do it.

Comment (arcanis) [reply packages/yarnpkg-core/sources/Project.ts:1887]: Do you feel strongly against it? I still think it'd be easier to read, and I'd imagine a nested Promise.all wouldn't have better perf than a spread.

Comment (merceyz) [reply packages/yarnpkg-core/sources/Project.ts]: No, I've committed the suggestion.

**Pull request merged**